src/fusor/fusion_matching.py

```diff
@@ -0,0 +1,281 @@
+"""Module for matching assayed fusions against categorical fusions"""
+
+import pickle
+from enum import Enum
+from pathlib import Path
+
+from civicpy import civic
+
+from fusor.harvester import CIVICHarvester
+from fusor.models import (
+    AssayedFusion,
+    CategoricalFusion,
+    GeneElement,
+    LinkerElement,
+    MultiplePossibleGenesElement,
+    TranscriptSegmentElement,
+    UnknownGeneElement,
+)
+from fusor.translator import Translator
+
+
+class FusionSources(str, Enum):
+    """Define CategoricalFusion sources"""
+
+    CIVIC = "CIViC"
+
+
+class FusionMatcher:
+    """Class for matching assayed fusions against categorical fusions"""
+
+    def __init__(self, translator: Translator):
+        """Initialize CategoricalFusion source"""
+        self.translator = translator
+        self.categorical_fusions: list[CategoricalFusion] = []
+
+    def load_categorical_fusions(self, output_dir: Path) -> list[CategoricalFusion]:
+        """Load in cache of CategoricalFusion objects
+
+        :param output_dir: A path to the cache
+        :return A list of Categorical fusions
+        """
+        with output_dir.open("rb") as categorical_fusions:
+            return pickle.load(categorical_fusions)  # noqa: S301
+        self.categorical_fusions = categorical_fusions
+        return self.categorical_fusions
```

Comment:
Wouldn't we want something like below?

```suggestion
    def __init__(self, translator: Translator, output_dir: Path):
        """Initialize CategoricalFusion source"""
        self.translator = translator
        self.categorical_fusions: list[CategoricalFusion] = self.load_categorical_fusions(output_dir)

    def load_categorical_fusions(self, output_dir: Path) -> list[CategoricalFusion]:
        """Load in cache of CategoricalFusion objects

        :param output_dir: A path to the cache
        :return A list of Categorical fusions
        """
        with output_dir.open("rb") as categorical_fusions:
            return pickle.load(categorical_fusions)  # noqa: S301
```

Reply:
@korikuzma I had an empty list get initialized in the event that the user was running the fusion matching module for the first time and had not yet generated a cached file of categorical fusions. In this case, running `load_categorical_fusions` in the `init` function would fail.